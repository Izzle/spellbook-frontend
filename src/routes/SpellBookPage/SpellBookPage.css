.SpellBookPage__container {
    width: 600px;
    margin: 0 auto;
}

.SpellBookPage__Spells__container {
    display: flex;
}