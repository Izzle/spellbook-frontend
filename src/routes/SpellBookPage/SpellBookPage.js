import React, { Component } from 'react';
import SpellContext from '../../contexts/SpellContext';
import SpellApiService from '../../services/spell-api-service';
import SpellLibraryOptions from '../../components/SpellLibraryOptions/SpellLibraryOptions';
import SpellBookButtons from '../../components/SpellBookButtons/SpellBookButtons';
import Spell from '../../components/Spell/Spell';
import './SpellBookPage.css';

export default class SpellBookPage extends Component {
    static defaultProps = {
        match: { params: {} }
    };

    static contextType = SpellContext;

    componentDidMount() {
        const { id } = this.props.match.params;
        this.context.clearError();
        // Gets the spells from DB and updates state to use that
        SpellApiService.getAllSpellsInSpellbookById(id)
          .then(this.context.setSpells)
          .catch(this.context.setError);
    }

    render() {
        const { spells = [] } = this.context;
        return (
            <section className='SpellBookPage'>
                <div className='SpellBookPage__container'>
                <SpellLibraryOptions />
                <SpellBookButtons />
                <div className='SpellBookPage__Spells__container'>
                    {spells.map((spell, idx) => {
                        return <Spell
                        key={idx}
                        id={spell.id}
                        name={spell.spell_name}
                        classes={spell.spell_classes /* This doesnt exist like this anymore, I had to make a table to JOIN tables called class_spells */}
                        level={spell.spell_level}
                        school={spell.spell_school}
                        range={spell.spell_range}
                        castingTime={spell.cast_time}
                        spellComponents={spell.spell_components}
                        duration={spell.spell_duration}
                        description={spell.spell_description}
                        higherLevels={spell.higher_levels}
                        />
                    })}
                </div>
                </div>
            </section>
        );
    }
}