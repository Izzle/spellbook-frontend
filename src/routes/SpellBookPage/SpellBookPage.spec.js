import React from 'react';
import renderer from 'react-test-renderer';
import ReactDOM from 'react-dom';
import SpellBookPage from './SpellBookPage';
import { MemoryRouter } from 'react-router-dom';


describe('<SpellBookPage />', () => {
    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<MemoryRouter><SpellBookPage /></MemoryRouter>, div);
        ReactDOM.unmountComponentAtNode(div);
    });
    it('renders this UI as expected', () => {
        const tree = renderer.create(<MemoryRouter><SpellBookPage /></MemoryRouter>).toJSON();
        expect(tree).toMatchSnapshot();
    });    
});