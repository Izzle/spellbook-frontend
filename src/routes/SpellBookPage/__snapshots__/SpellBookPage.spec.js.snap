// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SpellBookPage /> renders this UI as expected 1`] = `
<section
  className="SpellBookPage"
>
  <div
    className="SpellBookPage__container"
  >
    <h2 />
    <div>
      <a
        className="SpellBookButtons__Link"
        href="/spellbook/undefined/build"
        onClick={[Function]}
      >
        <button
          className="SpellBookButtons__button"
        >
          Edit Spellbook
        </button>
      </a>
      <a
        className="SpellBookButtons__Link"
        href="/spellbook"
        onClick={[Function]}
      >
        <button
          className="SpellBookButtons__button"
        >
          View All Spellbooks
        </button>
      </a>
    </div>
    <div
      className="SpellBookPage__Spells__container"
    />
  </div>
</section>
`;
